services/catalog/tests/test_authentication_catalog.py

```diff
@@ -241,4 +226,145 @@ async def sub_authentication(client, monkeypatch, httpx_mock: HTTPXMock):
             "stac_version": "1.0.0",
         },
     ]
-    assert content["collections"] == valid_collections
+    # Pass the api key by HTTP headers and by url query parameter
+    for pass_the_apikey in PASS_THE_APIKEY:
+        all_collections = client.request("GET", "/catalog/collections", **pass_the_apikey)
+
+        assert all_collections.status_code == HTTP_200_OK
+        content = json.loads(all_collections.content)
+        assert content["collections"] == valid_collections
+
+
+"""
+TODO for Youri, fix these tests
```

Comment:
@cyouri-csgroup I let you fix these tests. Also, you should implement tests that verify that we get an error status code if we pass no apikey or an invalid apikey (as in test_authentication.py)